CMakeLists.txt

```diff
@@ -13,21 +18,22 @@ set(main_exe_name RigelEngine)
 # Project settings
 ###############################################################################
 
-option(USE_GL_ES "Use OpenGL ES instead of regular OpenGL" OFF)
+option(USE_GL_ES "Use OpenGL ES instead of regular OpenGL" ON)
 option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
 
-
 # Dependencies
 ###############################################################################
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+# set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "/home/pratikone/git/emsdk/upstream/emscripten/cmake/Modules/")
 
-set(Boost_USE_STATIC_LIBS ON)
-find_package(Boost 1.65 COMPONENTS program_options REQUIRED)
-find_package(SDL2 REQUIRED)
-find_package(SDL2_mixer REQUIRED)
+# set(Boost_USE_STATIC_LIBS ON)
+# find_package(Boost 1.65 COMPONENTS program_options REQUIRED)
+# find_package(SDL2 REQUIRED)
+# find_package(SDL2_mixer REQUIRED)
 find_package(Filesystem REQUIRED)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -std=c++1z -s WASM=1 -s USE_BOOST_HEADERS=1 -s USE_SDL=2 -s USE_SDL_MIXER=2 -s ALLOW_MEMORY_GROWTH=1 -s MIN_WEBGL_VERSION=1 -s MAX_WEBGL_VERSION=2 -s FULL_ES2=1 --preload-file /home/pratikone/Desktop/duke/")
```

Comment:
I guess the preload files would be checked into the repo? Then it would be easy to make a relative path (using `${CMAKE_SOURCE_DIR}`)

Reply:
Although that brings up the question whether it would be ok to check a copy of the Duke Shareware files into the repo, which I'm not sure about. This would be avoided by adding the automatic download feature, of course.

Reply:
Yup. we are in the similar situation as OpenLara project. They have shipped some form of files when porting Tomb Raider to webassembly http://xproger.info/projects/OpenLara/

Reply:
Didn't know about that project! It's so cool to see how many of these reverse engineering projects there are nowadays :slightly_smiling_face: 

There's also the option to write a script `setup_wasm_build.sh`, which could take the path to the game files as an argument, run CMake with the right options, and then copy the game files into the build directory. That way, we can easily produce wasm builds with the game files included, but we don't have to check them into the repo.

Reply:
yeah, open source clone of Tomb Raider has existed since 2013 atleast. They are doing pretty good covering TR 1-5.